packages/docusaurus/src/commands/swizzle/common.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import leven from 'leven';
+import logger from '@docusaurus/logger';
+import type {
+  InitializedPlugin,
+  SwizzleAction,
+  SwizzleActionStatus,
+} from '@docusaurus/types';
+import _ from 'lodash';
+import type {NormalizedPluginConfig} from '../../server/plugins/init';
+
+export const SwizzleActions: SwizzleAction[] = ['wrap', 'eject'];
+
+export const SwizzleActionsStatuses: SwizzleActionStatus[] = [
+  'safe',
+  'unsafe',
+  'forbidden',
+];
+
+export const PartiallySafeHint = logger.red('*');
+
+export function actionStatusLabel(status: SwizzleActionStatus): string {
+  return _.capitalize(status);
+}
+
+const SwizzleActionStatusColors: Record<
+  SwizzleActionStatus,
+  (str: string) => string
+> = {
+  safe: logger.green,
+  unsafe: logger.yellow,
+  forbidden: logger.red,
```

Comment:
`forbidden` components should not be listed at all?

Reply:
a component might be forbidden to wrap but not to eject

if there's a table line displayed with both wrap/eject, we should probably display something in the table cell